packages/components/src/mobile/link-picker/link-picker-screen.native.js

```diff
@@ -18,14 +20,20 @@ const LinkPickerScreen = ( { returnScreenName } ) => {
 	const route = useRoute();
 
 	const onLinkPicked = ( { url, title } ) => {
-		navigation.navigate( returnScreenName, {
-			inputValue: url,
-			text: title,
-		} );
+		Keyboard.dismiss();
```

Comment:
I understand that this mirrors what we do on Android elsewhere ([here perhaps?](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/format-library/src/link/modal-screens/link-picker-screen.native.js#L23-L24)) and think that this is more than fine to include in the RN upgrade PR, but was wondering if we have thoughts about the fact that a component like the link-picker needs to wrangle the dismissal of the virtual keyboard 🤔. I'd kinda expect that operation to be handled more generally, perhaps by the navigation components?

Nothing to deal or refactor here, just wanted to grab your thoughts on this @dcalhoun , wdyt?

Reply:
> I understand that this mirrors what we do on Android elsewhere ([here perhaps?](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/format-library/src/link/modal-screens/link-picker-screen.native.js#L23-L24)) and think that this is more than fine to include in the RN upgrade PR, but was wondering if we have thoughts about the fact that a component like the link-picker needs to wrangle the dismissal of the virtual keyboard 🤔. I'd kinda expect that operation to be handled more generally, perhaps by the navigation components?

That is correct. The Android implementation originates from https://github.com/WordPress/gutenberg/commit/98684f134ddf562d815fc2c6711e3eefc75a7443 and https://github.com/WordPress/gutenberg/commit/f8b0275acd4c6693698f86d61095ae62a25696fd. 

I agree that expecting one-off components to manage keyboard dismissal to be cryptic and brittle. Ideally, we implement a system that appropriately manages keyboard visibility, navigation transitions, etc at a lower level. 

This particular change (https://github.com/WordPress/gutenberg/commit/134470fbd1a8f0189628cbfdcbb69b28f9d070b3) was done as a way to avoid unconsumed `LayoutAnimation` issues that caused non-dismissed, invisible modals on iOS. My understanding is that the original implementation, for Android only, was done to improve animation performance.

Reply:
Thank you for your thoughts David. Would you mind opening a ticket to capture the desire to refactor the keyboard dismissal so it's easier to remember and circle back to? Thanks!

Reply:
I have opened https://github.com/WordPress/gutenberg/issues/37559 to track this subject.